### PR TITLE
Remove column names

### PR DESCRIPTION
**Pull request by vuule**
## Description
Closes https://github.com/rapidsai/cudf/issues/6411

Removed `column_names` in favor of `schema_info`.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by galipremsagar — APPROVED** [review]

**Review by elstehle — APPROVED** [review]
C++ changes LGTM

**Review by karthikeyann — APPROVED** [review]
Looks good to me.

**Review by bdice — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

File: cpp/include/cudf/io/types.hpp [hunk 1]
```diff
@@ -144,7 +144,6 @@ struct column_name_info {
  * f5    f6
  */
 struct table_metadata {
-  std::vector<std::string> column_names;  //!< Names of columns contained in the table
```
[the remark this answers is not in the document]
Comment (vuule) [reply cpp/include/cudf/io/types.hpp:147]: good catch! removed the outdated docs

**Pull request merged**